### ISSUE TITLE
Accessibility issue: "clr-dg-action-overflow" open context menu on "Tab" key. It should open context menu on "Enter" key

### ISSUE THREAD

**Issue by sachinkkarande** [vmware/clarity#5924]
## Describe the bug
Accessibility issue: "clr-dg-action-overflow" open context menu on "Tab" key. It should open context menu on "Enter" key

## How to reproduce

Use "Tab" key to navigate through the grid. "Tab" key on the 1st row 1st column opens context menu. (Context menu should open on "Enter" key and not on "Tab" key).
Observe that user is not able to move to 2nd row by using only the keyboard as the context menu of the 1st row opens as soon as "Tab" is clicked.

Editor URL: 
https://stackblitz.com/edit/clarity-app-modal-datagrind-context-menu-5hdjw7?file=src/app/main/main.component.html

Application URL:
https://clarity-app-modal-datagrind-context-menu-5hdjw7.stackblitz.io

1.  Go to application
2.  try to navigate the grid using the "Tab" key
3.  Observe that "Tab" key opens context menu on 1st row and 1st column, Esc key will close the dialog but again pressing "Tab" key opens same context menu.
4.  Use cannot move beyond 1st row 1st column by using tab key

* Violation of Accessibility Standards:  
WCAG 2.1 -> 2.1.2 No Keyboard Trap (Level A)

## Expected behavior

When tab to the available actions button, don't expand the Edit/Delete/Copy items automatically, only expanded it when user press Enter key on available actions button.
And after tab to the last item, the focus can move out to the component which after the available actions button.

## Versions

* For version "4.0.3" and "5.1.1" issue is observed.

     /* "Tab" key on the first Grid column opens context menu created by <clr-dg-action-overflow> */
    "@clr/angular": "4.0.3",
    "@clr/core": "^3.0.0",
    "@clr/icons": "4.0.3",
    "@clr/ui": "4.0.3",
    
    /* "Tab" key on the first Grid column opens context menu created by <clr-dg-action-overflow> */ 
    "@clr/angular": "5.1.1",
    "@clr/core": "^3.0.0",
    "@clr/icons": "5.1.1",
    "@clr/ui": "5.1.1",

**Device:**

Issue is not OS / Browser specific

## Additional notes

Check screens recording: 
"clr-dg-action-overflow opens context menu on TAB key 5-May-21.mov"

https://user-images.githubusercontent.com/83701351/117134526-4f900a80-adc3-11eb-8ce7-7fb71b0feac9.mov

**Comment by jeeyun** [comment]
Hello,
Thank you for filing an issue. It looks like something is wrong in your stackblitz set up as I'm not seeing the overflow button render correctly. 

In addition, in order for your datagrid row to have access to the data object you should use `[clrDgItem]` in your datagrid as shown in this example: https://v4.clarity.design/documentation/datagrid/single-action

That being said the version you mentioned (4.0.3) does have the keyboard navigation issue and seems like we have fixed the bug since then. I tested this with the latest 4.x version (4.0.14) and it's working as expected:
https://stackblitz.com/edit/clarity-eljv7g?file=app%2Fapp.component.ts

Try upgrading your clarity version to latest (we follow sem ver so there is no breaking changes!) 

Closing this issue as it's working as expected.
